usaspending_api/awards/v2/data_layer/orm.py

```diff
@@ -370,8 +368,77 @@ def fetch_business_categories_by_legal_entity_id(legal_entity_id):
 
 
+def fetch_all_cfda_details(award):
+    # we default to fain because we don't have a good link between award -> fabs
+    # if the award doesn't have fain, use uri, if no uri, use latest transaction
+    filter = {"fain": award["fain"]}
+    if award["fain"] is None:
+        if award["uri"] is None:
+            filter = {"pk": award["_trx"]}
+        else:
+            filter = {"uri": award["uri"]}
+    queryset = TransactionFABS.objects.filter(**filter).values(
+        "cfda_number", "federal_action_obligation", "non_federal_funding_amount", "total_funding_amount"
+    )
+    cfdas = {}
+    for item in queryset:
+        # sometimes the transactions data has the trailing 0 in the CFDA number truncated, this adds it back
+        num = re.sub(r"^\d*\.\d\d$", "\\g<0>0", item["cfda_number"])
+        if cfdas.get(num):
+            cfdas.update(
+                {
+                    num: {
+                        "federal_action_obligation": float(cfdas[num]["federal_action_obligation"] or 0)
+                        + float(item["federal_action_obligation"] or 0),
+                        "non_federal_funding_amount": float(cfdas[num]["non_federal_funding_amount"] or 0)
+                        + float(item["non_federal_funding_amount"] or 0),
+                        "total_funding_amount": float(cfdas[num]["total_funding_amount"] or 0)
+                        + float(item["total_funding_amount"] or 0),
+                    }
+                }
+            )
+        else:
+            cfdas.update(
+                {
+                    num: {
+                        "federal_action_obligation": float(item["federal_action_obligation"] or 0),
+                        "non_federal_funding_amount": float(item["non_federal_funding_amount"] or 0),
+                        "total_funding_amount": float(item["total_funding_amount"] or 0),
+                    }
+                }
+            )
+
+    c = []
+    for key in cfdas.keys():
+        details = fetch_cfda_details_using_cfda_number(key)
+        if details.get("url") == "None;":
+            details.update({"url": None})
+        c.append(
+            {
+                "cfda_number": key,
+                "federal_action_obligation_amount": cfdas[key]["federal_action_obligation"],
+                "non_federal_funding_amount": cfdas[key]["non_federal_funding_amount"],
+                "total_funding_amount": cfdas[key]["total_funding_amount"],
+                "cfda_title": details.get("program_title"),
+                "cfda_popular_name": details.get("popular_name"),
+                "cfda_objectives": details.get("objectives"),
+                "cfda_federal_agency": details.get("federal_agency"),
+                "cfda_website": details.get("website_address"),
+                "sam_website": details.get("url"),
+                "cfda_obligations": details.get("obligations"),
+            }
+        )
+    c.sort(key=lambda x: x["total_funding_amount"], reverse=True)
```

Comment:
Glad you incorporated this from a recent jira comment. It might be good to add a unit test to verify the sort order against future changes

Reply:
added test in `usaspending_api/awards/tests/test_awards_v2.py`